src/Tasks/Microsoft.NET.Build.Tasks/PublishAssembliesResolver.cs

```diff
@@ -31,6 +31,11 @@ public IEnumerable<ResolvedFile> Resolve(ProjectContext projectContext)
 
             foreach (LockFileTargetLibrary targetLibrary in projectContext.GetRuntimeLibraries(_privateAssetPackageIds))
             {
+                if (targetLibrary.Type == "project")
```

Comment:
Turns out that libraries of type `project` are already handled specially for compile items: https://github.com/dotnet/sdk/blob/7baf434d67f8bce169755c46fb3e6d474c5fd521/src/Tasks/Microsoft.NET.Build.Tasks/DependencyContextBuilder.cs#L378-L382